components/posts/posts.php

```diff
@@ -29,29 +29,34 @@ function mm_posts( $args ) {
 		'taxonomy'            => '',
 		'term'                => '',
 		'heading_level'       => 'h1',
-		'per_page'            => 10,
+		'per_page'            => '',
 		'pagination'          => '',
 		'template'            => '',
 		'show_featured_image' => false,
 		'featured_image_size' => 'thumbnail',
 		'show_post_info'      => false,
 		'show_post_meta'      => false,
 		'use_post_content'    => false,
+		'ajax_filter'         => false,
+		'filter_style'        => 'links',
 		'link_title'          => true,
 		'masonry'             => false,
 	);
 	$args = wp_parse_args( (array)$args, $defaults );
 
 	// Get clean param values.
-	$query_type    = sanitize_text_field( $args['query_type'] );
-	$post_ids      = $args['post_ids'] ? str_getcsv( $args['post_ids'] ) : '';
-	$post_type     = sanitize_text_field( $args['post_type'] );
-	$taxonomy      = sanitize_text_field( $args['taxonomy'] );
-	$term          = sanitize_text_field( $args['term'] );
-	$heading_level = sanitize_text_field( $args['heading_level'] );
-	$per_page      = (int)$args['per_page'];
-	$template      = sanitize_text_field( $args['template'] );
-	$masonry       = mm_true_or_false( $args['masonry'] );
+	$query_type      = sanitize_text_field( $args['query_type'] );
+	$post_ids        = $args['post_ids'] ? str_getcsv( $args['post_ids'] ) : '';
+	$post_type       = sanitize_text_field( $args['post_type'] );
+	$taxonomy        = sanitize_text_field( $args['taxonomy'] );
+	$term            = sanitize_text_field( $args['term'] );
+	$heading_level   = sanitize_text_field( $args['heading_level'] );
```

Comment:
Technically you want different handling for different use cases here. $args that are about to be used as args for `WP_Query` should get the `sanitize_text_field()` treatment, while args destined for display should get some `esc_*()` treatment but those should be done as late as possible, so most/all of those can be omitted here and escaped later as they are being output.